src/games/strategy/triplea/util/TransportUtils.java

```diff
@@ -198,4 +213,18 @@ public int compare(final Unit o1, final Unit o2) {
     return Optional.empty();
   }
 
+  private static void loadMaxUnits(final Unit transport, final List<Unit> canBeTransported,
```

Comment:
This is a bit procedural. The data state of `mapping` is being controlled by code that does not own the `mapping` object. So that is the problem that one would want to fix. In this case we would have pure in parameters and an out parameter. This way the data is still owned by the client of this method, we just compute a value for them here.

Another notable thing is that this code is repeated:
`final int capacity = TransportTracker.getAvailableCapacity(transport);`
That makes me think it has been sliced up not so cleanly.

I think the way to go is to get the first unit with cost less then capacity. Then have the client code deal with updating the map.

Something like this:
`canBeTransported.stream().filter( unit -> UnitAttachment.get((unit).getType()).getTransportCost() <= capacity ).getFirst()`

That returns an optional, which would then play well with the optional above.